python/pyhail/dataset.py

```diff
@@ -117,15 +117,114 @@ def aggregate_by_key(self, key_code, agg_code):
+        **Examples**
+
+        Calculate the total number of SNPs, indels, and variants contained in
+        the intervals specified by *data/capture_intervals.txt*:
+
+        >>> vds.aggregate_intervals('data/capture_intervals.txt',
+        >>>   """n_SNP = variants.filter(v => v.altAllele.isSNP).count(),
+        >>>      n_indel = variants.filter(v => v.altAllele.isIndel).count(),
+        >>>      n_total = variants.count()"""
+        >>>   'out.txt')
+
+        If *data/capture_intervals.txt* contains::
+
+            4:1500-123123
+            5:1-1000000
+            16:29500000-30200000
+
+        then the previous expression writes something like the following to
+        *out.txt*::
+
+            Contig    Start       End         n_SNP   n_indel     n_total
+            4         1500        123123      502     51          553
+            5         1           1000000     25024   4500        29524
+            16        29500000    30200000    17222   2021        19043
+
+        The parameter ``condition`` defines the names of the column headers (in
+        the previous case: ``n_SNP``, ``n_indel``, ``n_total``) and how to
+        calculate the value of that column for each interval.
+
+        Count the number of LOF, missense, and synonymous non-reference calls
+        per interval:
+
+        >>> (vds.annotate_variants_expr('va.n_calls = gs.filter(g.isCalledNonRef).count()')
+        >>>     .aggregate_intervals('data/intervals.txt'
+        >>>        """LOF_CALLS = variants.filter(v => va.consequence == "LOF").map(v => va.n_calls).sum(),
+        >>>           MISSENSE_CALLS = variants.filter(v => va.consequence == "missense").map(v => va.n_calls).sum(),
+        >>>           SYN_CALLS = variants.filter(v => va.consequence == "synonymous").map(v => va.n_calls).sum()"""
+        >>>        'out.txt'))
+
+        If *data/intervals.txt* contains::
+
+            4:1500-123123
+            5:1-1000000
+            16:29500000-30200000
+
+        then the previous expression writes something like the following to
+        *out.txt*::
+
+            Contig    Start       End         LOF_CALLS   MISSENSE_CALLS   SYN_CALLS
+            4         1500        123123      42          122              553
+            5         1           1000000     3           12               66
+            16        29500000    30200000    17          22               202
+
+        **Notes**
+
+        Intervals are **left inclusive, right exclusive**.  This means that
+        [chr1:1, chr1:3) contains chr1:1 and chr1:2.
+
+        **Designating output with an expression**
+
+        An export expression designates a list of computations to perform, and
+        what these columns are named.  These expressions should take the form
+        ``COL_NAME_1 = <expression>, COL_NAME_2 = <expression>, ...``.
+
+        The ``condition`` parameter has the following namespace:
+
+        +--------------+------------------------------------------+
```

Comment:
Look at current version of annotate_samples_expr for how to format this. Also can link to interval like this
:ref:\`interval\`